### PR TITLE
Edit rest of document (The empty interface and Summary)

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
This PR is a follow-up to #12 that edits "The empty interface{}" and "Summary."

Please let me know if you have any questions or would like to request changes. Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1213,46 +1213,48 @@ func TestFn(t *testing.T) {
...
+As mentioned in a previous section, Go determines whether a concrete type implements a particular interface by checking whether the type implements the <em>methods</em> of that interface. So what happens if our interface declares no methods, as is the case with the empty interface?
 
 ```go
 type EmptyInterface interface {}
 ```
 
-The above being equivalent to the built-in type `interface{}`. The result of this interface type is that **any** type is accepted. Meaning, that we can write functions in which any type is accepted. This is super useful for certain kind of functions, such as when creating a printer function. This is how it's possible to give any type to the `Println` function from the `fmt` package:
+The above is equivalent to the built-in type `interface{}`. A natural consequence of this is that we can write generic functions&mdash;ones that accept any type as arguments. This is extremely useful for certain kinds of functions, such as print helpers. Interestingly, this is actually what makes it possible to pass in any type to the `Println` function from the `fmt` package:
```
Comment (Pungyeon) [inline README.md]:
```suggestion
The above is equivalent to the built-in type `interface{}`. A natural consequence of this is that we can write generic functions that accept any type as arguments. This is extremely useful for certain kinds of functions, such as print helpers. Interestingly, this is actually what makes it possible to pass in any type to the `Println` function from the `fmt` package:
```

File: README.md [hunk 2]
```diff
@@ -1213,46 +1213,48 @@ func TestFn(t *testing.T) {
...
 
 ```go
 func Println(v ...interface{}) {
     ...
 }
 ```
 
-In this case, we aren't only accepting a single `interface{}` but rather, a slice of types. These types can be of any type and can even be of different types, as long as they implement the empty `interface{}`, which we are certain that any type will. This is a super common pattern when handling string conversation (both from and to string). The reason being, this is the only way in Go to implement generic methods. Good examples of this, come from the `json` standard library package:
+In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types. These can be <em>any</em> types, not necessarily all the same. The only requirement is that all of the types implement the empty `interface{}`, which we are certain that any type will. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```
Comment (Pungyeon) [inline README.md]: One of those times where I feel that a sentence should be a little different, but not sure, if below is an improvement?

```suggestion
In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types. The only requirement is that all of the types implement the empty `interface{}`, which we are certain that any type will. This enables `Println` to accept a slice of <em>any</em> type, whose elements aren't restricted to being uniform. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

File: README.md [hunk 3]
```diff
@@ -1213,46 +1213,48 @@ func TestFn(t *testing.T) {
...
+        http.Error(w, err.Error(), http.StatusInternalServerError)
+        return
+    }
+    w.WriteHeader(http.StatsOK)
 }
 ```
 
-All the *less elegant* code, is contained within the  `Decode` function. Developers using this functionality, therefore, won't have to worry about reflection or casting of types. We just have to worry about providing a pointer to a concrete type. This is good, because the `Decode()` function is, technically, returning a concrete type. We are passing in our `Item` value, which will be populated from body of the http request and we won't have to deal with the potential risks of handling the `interface{}` value.
+All the less elegant code is hidden within the `Decode` function. Thus, developers using this functionality won't have to worry about type reflection or type casting; we just have to worry about providing a pointer to a concrete type. This is good because the `Decode()` function is technically returning a concrete type. We are passing in our `Item` value, which will be populated from the body of the HTTP request. This means we won't have to deal with the potential risks of handling the `interface{}` value ourselves.
```
Comment (Pungyeon) [inline README.md]: So, my choice of the word 'contained' was actually rather specific.... I'm not really trying to hide the implementation (necessarily), I'm just trying to isolate / contain it, so users of the library aren't affected.

**Review by Pungyeon — CHANGES_REQUESTED** [review]
Really good work again @AleksandrHovhannisyan , you are an absolute machine 💪 

Just a few pointers, but other than that it's perfect 👌

File: README.md (hunk 1 above)
Comment (AleksandrHovhannisyan) [reply README.md]: Good point, that actually reads better! Could you also get rid of the `&mdash;` before it and replace it with a space?

File: README.md (hunk 3 above)
Comment (AleksandrHovhannisyan) [reply README.md]: Ah, my bad! Let's go with contained. Do you want to make the suggestion or should I push an edit on my end?

File: README.md (hunk 2 above)
Comment (AleksandrHovhannisyan) [reply README.md]: Hmm, I think we end up repeating `the function accepts a <em>slice</em> of types` when we say `This enables `Println` to accept a slice of <em>any</em> type, whose elements aren't restricted to being uniform.`

I do agree that my edit is a bit wordy, though. What do you think of this alternative?

```
According to the signature, `Println` accepts a single `interface{}`. In reality, this means it accepts a <em>slice</em> of any type, whose elements need not be uniform. This is a very common pattern...
```

File: README.md (hunk 1 above)
Comment (Pungyeon) [reply README.md]: done!

File: README.md (hunk 3 above)
Comment (Pungyeon) [reply README.md]: Just push on your end :)

File: README.md (hunk 2 above)
Comment (Pungyeon) [reply README.md]: Hmm... how about this instead then?

```suggestion
In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types that implement the empty `interface{}`. As there are no methods associated with the empty `interface{}` <em>all</em> types are accepted, even making it possible to feed `Println` with a slice of different types. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

Comment (AleksandrHovhannisyan) [reply README.md]: Yep, that works! Just be sure to add a comma after ``interface{}`` :)

Comment (Pungyeon) [reply README.md]: I'm guessing you mean before the <em>all</em>?

```suggestion
In this case, `Println` isn't just accepting a single `interface{}`; rather, the function accepts a <em>slice</em> of types that implement the empty `interface{}`. As there are no methods associated with the empty `interface{}`, <em>all</em> types are accepted, even making it possible to feed `Println` with a slice of different types. This is a very common pattern when handling string conversation (both from and to a string). Good examples of this come from the `json` standard library package:
```

File: README.md (hunk 3 above)
Comment (AleksandrHovhannisyan) [reply README.md]: Done!

File: README.md (hunk 2 above)
Comment (AleksandrHovhannisyan) [reply README.md]: Oh, yeah, there :)

**Review by Pungyeon — APPROVED** [review]
Thank you so much again for your awesome work @AleksandrHovhannisyan I'm so appreciative for all of the time you must be spending on this 🙏 You are an absolute hero  🎉

**Comment by AleksandrHovhannisyan** [comment]
No problem! Truth be told, I didn't have much else to do 😄
